source/_integrations/apple_tv.markdown

```diff
@@ -10,165 +10,229 @@ ha_iot_class: Local Push
+
+To add one or more Apple TVs to your system, add the following to your `configuration.yaml` file:
 
 ```yaml
 # Example configuration.yaml entry
 apple_tv:
-  - host: IP_1
-    login_id: LOGIN_ID_1
-    name: NAME_1
-    start_off: START_OFF_1
-    credentials: CREDENTIALS_1
-  - host: IP_2
-    login_id: LOGIN_ID_2
-    name: NAME_2
-    start_off: START_OFF_2
-    credentials: CREDENTIALS_2
+  - address: IP
+    name: NAME
+    identifier: ID
+    protocol: PROTOCOL
+    start_off: START_OFF
+    credentials:
+      dmap: DMAP_CREDENTIALS
+      mrp: MRP_CREDENTIALS
+      airplay: AIRPLAY_CREDENTIALS
 ```
 
 {% configuration %}
-host:
-  description: The IP-address of the device.
-  required: true
-  type: string
-login_id:
-  description: An identifier used to login to the device, see below.
+address:
+  description: The IP address of the device.
   required: true
   type: string
 name:
   description: The name of the device used in the frontend.
   required: false
   type: string
+identifier:
+  description: A unique identifier used to identify the device, see instructions below.
+  required: true
+  type: string
+protocol:
+  description: Protocol used to connect, either DMAP or MRP. Credentials must be provided for the selected protocol. See instructions below.
+  required: true
+  type: string
 start_off:
   description: Set to true if the device should start in fake standby.
   required: false
   type: boolean
   default: false
 credentials:
-  description: Credentials used for AirPlay playback.
-  required: false
-  type: string
+  type: map
+  description: Credentials for protocols.
+  keys:
+    dmap:
+      description: Credentials used for DMAP protocol (Apple TV 3 and earlier).
+      required: false
+      type: string
+    mrp:
+      description: Credentials used for AirPlay protocol (Apple TV 4 or later).
+      required: false
+      type: string
+    airplay:
+      description: Credentials used for AirPlay protocol.
+      required: false
+      type: string
 {% endconfiguration %}
 
-In order to connect to the device, you need a *login id*. The easiest way to obtain this identifier is to use the `apple_tv_scan` service (described below). Additional information about `start_off` and `credentials` can also be found under the guides section.
-
-## Guides
-
-### Scanning for devices
-
-Make sure Home Sharing is enabled on the Apple TV.
-
-To scan for devices and determine the `login_id`, open the developer tools by selecting the hammer icon in the sidebar. Once in the developer tools select **services**.
-
-<img src='/images/screenshots/developer-tools.png' />
-
-Select `apple_tv` as domain and `apple_tv_scan` as service then press the button:
-
-<img src='/images/integrations/apple_tv/scan_start.jpg' />
-
-Scanning will be done for three seconds and notification will be shown in the state view with all found devices:
+To find `identifier` you can use the `atvremote` command, which is bundled with `pyatv` (the library
```

Comment:
This is problematic for Home Assistant & Home Assistant Core on Docker users?

Reply:
Everyone is supposed to use the config flow (which hides all of this). Only advanced users should should (possibly) use this and they should be capable of setting this up. You don't have to run the command on the same machine you run Home Assistant, it can be any machine on the network (pyatv is supported on Windows, Linux and macOS).

Reply:
Yeah, so, maybe it should be a bit clearer on that in the documentation... that was kinda my point. :)

Reply:
Yeah, I can agree with you 😊 I added a mild "warning" in the beginning of the section. Have a look.